StardewArchipelago/ModEntry.cs

```diff
@@ -125,6 +128,32 @@ private void OnGameLaunched(object sender, GameLaunchedEventArgs e)
             ResetArchipelago();
         }
 
+        private void OnMenuChanged(object sender, MenuChangedEventArgs e)
+        {
+            if (e.NewMenu is CarpenterMenu)

```

Comment:
Could you elaborate a bit?  Just want to make sure before make this part of the code better.

Reply:
```
if (e.NewMenu is not CarpenterMenu carpenterMenu)
{
    return;
}
var blueprints = _helper.Reflection
.GetField<List<BluePrint>>(carpenterMenu, "blueprints")
.GetValue();
```

This is because e.NewMenu is an IClickableMenu. You are immediately leaving the method if it isn't a CarpenterMenu, which is fine, and later on you rightfully expect it to be a CarpenterMenu to grab it's "blueprints" field. This is all logically sound. But getting the blueprints is done with reflection, so the type of the variable is irrelevant, it just looks for the field at runtime. This essentially bypasses the whole idea of typing things correctly (and is the main reason why reflection is frowned upon, although when modding we often don't have a choice.)

This way, you cast your item to the correct type before getting blueprints, so it's really obvious that you are going for a variable that has the field and is less error prone in the future.